lib/elixir/lib/calendar.ex

```diff
@@ -265,6 +265,14 @@ defmodule Calendar do
   """
   @callback valid_time?(hour, minute, second, microsecond) :: boolean
 
+  @doc """
+  Implements inspect for dates, datetimes and naive_datetimes.
+
+  The implementation must define inspect for `Date.t()`, `DateTime.t()` and
+  `NaiveDateTime.t()` structs.
+  """
+  @callback inspect(date | datetime | naive_datetime, Inspect.Opts.t()) :: String.t()
```

Comment:
We should also pass `time` in here. :)

Reply:
Done. Also added the implementation for `inspect/2` for time into `Calendar.ISO`. Unlike `Date`, `NaiveDateTime` and `DateTime` its not guaranteed that there is a `:calendar` field. So for time it matches on `%{calendar: calendar}` and defaults ` Inspect.Any.inspect/2` otherwise.

Reply:
Time should always have a calendar field. :) It is not guaranteed for something like CDLR, which may work with different Elixir versions, but in Elixir itself it will always be there. :)

Reply:
I think this is still unresolved:

```suggestion
  @callback inspect(time | date | datetime | naive_datetime, Inspect.Opts.t()) :: String.t()
```